doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,621 @@
+---
+title: "Introduction to igraph"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Introduction to igraph}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Tutorial
+
+The main goals of the igraph library is to provide a set of data types and functions for 1) pain-free implementation of graph algorithms, 2) fast handling of large graphs, with millions of vertices and edges, 3) allowing rapid prototyping via high level languages like R.
+
+## Creating a graph
+
+The simplest way to create a graph is the `Graph` constructor. To make an empty graph:
+
+```{r}
+library("igraph")
+
+gi <- make_empty_graph()
+```
+
+To make a graph with 10 nodes (numbered `1` to `10`) and two edges connecting nodes `1-2` and `1-5`:
+
+```{r}
+gi <- graph(edges = c(1,2, 1,5), n=10, directed = FALSE)
+```
+
+We can print the graph to get a summary of its nodes and edges:
+
+```{r echo = TRUE}
+print(gi)
+```
+
+This means: **U** ndirected graph with **10** vertices and **2** edges, with the exact edges listed out. If the graph has a [name] attribute, it is printed as well.
+
+Note
+
+`summary` is similar to `print` but does not list the edges, which is convenient for large graphs with millions of edges:
+
+```{r echo = TRUE}
+summary(gi)
+```
+
+## Adding/deleting vertices and edges
+
+Let's start from the empty graph again. To add vertices to an existing graph, use `add_vertices`:
+
+```{r}
+gi <- make_empty_graph()
+
+gi <- add_vertices(gi,3)
+```
+
+In igraph, vertices are always numbered up from one. The number of a vertex is called the *vertex ID*. A vertex might or might not have a name.
+
+Similarly, to add vertices use `add_edges`:
+
+```{r}
+gi <- add_edges(gi, edges = c(1,2, 1,3))
+```
+
+Edges are added by specifying the source and target vertex for each edge. This call added two edges, one connecting vertices `1` and `2`, and one connecting vertices `1` and `3`. Edges are also numbered up from one (the *edge ID*) and have an optional name.
+
+**Warning**
+
+Creating an empty graph and adding vertices and edges as shown here can be much slower than creating a graph with its vertices and edges as demonstrated earlier. If speed is of concern, you should especially avoid adding vertices and edges *one at a time*. If you need to do it anyway, you can use `add_vertex` and `add_edge`.
+
+If you try to add edges to vertices with invalid IDs (i.e., you try to add an edge to vertex `5` when the graph has only three vertices), you get an `igraph.InternalError` exception:
+
+```{r echo = TRUE, eval = FALSE}
+gi <- add_edges(gi, edges = c(5,4))
+```
+
+The message tries to explain what went wrong (`cannot add edges. -- Invalid vertex id`).
+
+Let us add some more vertices and edges to our graph:
+
+```{r echo = TRUE}
+gi <- add_edges(gi, edges = c(3,1))
+gi <- add_vertices(gi, 3)
+gi <- add_edges(gi, edges = c(3,4, 4,5, 5,6, 6,4))
+print(gi)
+```
```

Comment:
Unrelated to @szhorvat 's comment, but I think this would be a great place to introduce the user to the `magrittr` "pipe" syntax, which fits the use-case of chaining many graph-mutating operations:

```
gi <- gi %>% add_edges(edges=c(3,1)) %>% add_vertices(3) %>% add_edges(edges=c(3,4, 4,5, 5,6, 6,4))
```

Reply:
uuush this looks disgusting but if that's what people do... let's go with the people.